### DISCUSSION TITLE
Live-stream to AWS IVS via QR Control not working

### DISCUSSION THREAD

**Discussion by FoxBravoNL** [gopro/labs#222] [Q&A]
Could anyone point me in the right direction why livestreaming to Amazon IVS is not working through the QR code setup but _is_ working when using the Quick app?

I've set everything according to the GoPro Labs instructions including the full rtmps stream url including stream key, however when I launch the stream via the QR code it says "scanning for network" and then "button to cancel", so it seems like it's streaming.. but nothing is showing up. When I go through the same process using the Quick app it is working however, so the stream url + stream key are correct. 

Is there anything different about using the QR control way compared to the Quick app way?

I'm experimenting this setup for a product we're developing.. launching through the QR code would be a great benefit.

Any help is much appreciated.

**Comment by dnewman-gpsw** [comment]
What does the log file report?  On the SD Card in MISC/qrlog.txt

**Comment by FoxBravoNL** [comment]
Thanks for the quick reply, I've just started experimenting with the beta firmware so I didn't even know about the log file. :)

Here is the output relating to the stream:

[2023-01-30 21:40:00] Read QR:(secure RTMP) 
[2023-01-30 21:39:12] joining network:SFWifi 
[2023-01-30 21:54:52] LS State 1:20 
[2023-01-30 21:55:40] LS User Cancel:Button press 
[2023-01-30 21:55:40] LS Error:Live-streaming not start 

[2023-01-30 21:55:42] Read QR:oW1mVr1080!W!GL 
[2023-01-30 21:55:47] joining network:SFWifi 
[2023-01-30 21:55:50] LS State 1:20 
[2023-01-30 21:56:01] LS User Cancel:Button press 
[2023-01-30 21:56:01] LS Error:Live-streaming not start

**Reply by dnewman-gpsw** [reply]
I have not used AWS IVS end point before.  I just testing with a HERO11 out to twitch.com and it was working, the log file:

[2023-01-30 13:42:16] Read QR:(secure RTMP) 
[2023-01-30 13:42:20] Read QR:oW1mVr1080!W!GL 
[2023-01-30 13:42:28] joining network:JASADA2 
[2023-01-30 13:42:32] LS State 1:20 
[2023-01-30 13:42:33] LS State 1:17 
[2023-01-30 13:42:35] LS State 2:18 
[2023-01-30 13:42:39] LS Setup Complete:LIVE-STREAMING NOW! 

However Twitch is using rtmp:// not rtmps:// and the Labs hack may not support secure contentions.  This hack is a few years old now, originally developed on HERO8, before rtmps:// was common.

**Reply by FoxBravoNL** [reply]
I see.. that would make sense, thanks again for your help. I'm the co-founder of a company developing tools for aviation training ([VRef Solutions](https://vrefsolutions.com/product)) and currently we are using pretty big streaming cameras to fit our purpose. I was curious to see if GoPros would be able to fulfil our needs with all these extra possibilities. 

Is there any way to edit these hacks to make it compatible with rtmps do you think?

**Reply by dnewman-gpsw** [reply]
Yes, the Labs Live-streaming might be updated to support RTMPS, if that is the issue.  Can you try RTMP?  Are there any public AWS IVS end points we can test with?  Also try twitch.com to confirm the camera is behaving correctly.

**Reply by FoxBravoNL** [reply]
Awesome. I'll do some more tests including Twitch later this week. As far as a public endpoint for IVS... I couldn't find one easily but I'll have another look later.

**Reply by andrerav** [reply]
@FoxBravoNL If all else fails, you might be able to set up a reverse proxy with nginx that can forward the rtmp stream to your rtmps endpoint. Check out this post on SO: https://stackoverflow.com/questions/65474980/how-to-use-nginx-for-proxy-rtmp

**Reply by dnewman-gpsw** [reply]
I have found the issue, RTMPS requires the SSL certificates to be transmitted from the app. The camera does not have the certificates, so Labs alone can't communicate with secure connections.

**Reply by FoxBravoNL** [reply]
Thanks for the input @andrerav and @dnewman-gpsw. I think considering the limitations with regards to the SSL certificates that probably for our purposes it's not a suitable solution at this point but perhaps it's worth exploring further in the future.
